app/services/vacancy_facets.rb

```diff
@@ -2,7 +2,7 @@ class VacancyFacets
   CACHE_DURATION = 24.hours
 
   def job_roles
-    cached(:job_roles) { sort_and_limit(job_role_facet.select { |role| role.in?(Vacancy.primary_job_role_options) }, 4) }
+    cached(:job_roles) { job_role_facet.select { |role, count| role.in?(Vacancy.primary_job_role_options) && count.positive? } }
```

Comment:
We can still `sort_and_limit` though, in order to have the most popular listed first. Also, `sort_and_limit` checks whether count is positive. We just need to set `number_of_results` to the number of primary_job_role_options.